### PR TITLE
Adds offset '1', because pyrights offsets are zero-based (issue #25)

### PR DESCRIPTION
**Pull request by gersmann**

### CODE REVIEW

**Comment by jordemort** [comment]
Thanks for the fix! I checked out this branch and poked at it a bit. Based on pyright's output, and on the response to the issue you linked in the pyright repo, I think the column numbers are 0-based as well, and we need to bump those also.

Here's a raw diagnostic from pyright running on bad.py: 

```
        {
            "file": "/home/jordan/Source/action-pyright/pyright_to_rdjson/tests/files/bad.py",
            "severity": "error",
            "message": "Operator \"+\" not supported for types \"str\" and \"int\"\n  Operator \"+\" not supported for types \"str\" and \"int\"",
            "range": {
                "start": {
                    "line": 1,
                    "character": 11
                },
                "end": {
                    "line": 1,
                    "character": 20
                }
            },
            "rule": "reportGeneralTypeIssues"
        },
```

Here's the line it's referring to:

```
    return one + two + three
```

If you assume `character` is zero-based, then the string it calls out is `one + two`.

If you assume `character` is one-based, then the string it calls out is ` one + tw`.

According to the Reviewdog JSON Schema, their positions are one-based: https://github.com/reviewdog/reviewdog/blob/master/proto/rdf/jsonschema/Position.jsonschema

File: pyright_to_rdjson/pyright_to_rdjson.py [hunk 1]
```diff
@@ -33,11 +33,12 @@ def pyright_to_rdjson(jsonin: TextIO):
                     "path": d["file"],
                     "range": {
                         "start": {
-                            "line": d["range"]["start"]["line"],
+                            # pyright uses zero-based offsets
+                            "line": d["range"]["start"]["line"] + 1,
                             "column": d["range"]["start"]["character"],
```
Comment (jordemort) [inline pyright_to_rdjson/pyright_to_rdjson.py:38]:
```suggestion
                            "column": d["range"]["start"]["character"] + 1,
```

File: pyright_to_rdjson/pyright_to_rdjson.py [hunk 2]
```diff
@@ -33,11 +33,12 @@ def pyright_to_rdjson(jsonin: TextIO):
...
                         },
                         "end": {
-                            "line": d["range"]["end"]["line"],
+                            "line": d["range"]["end"]["line"] + 1,
                             "column": d["range"]["end"]["character"],
```
Comment (jordemort) [inline pyright_to_rdjson/pyright_to_rdjson.py:42]:
```suggestion
                            "column": d["range"]["end"]["character"] + 1,
```

File: pyright_to_rdjson/tests/test_pyright_to_rdjson.py [hunk 3]
```diff
@@ -43,8 +43,9 @@ def test_pyright_to_rdjson(self):
             pyright_line_start = pyright_line_range["start"]
             pyright_line_end = pyright_line_range["end"]
 
-            assert rdjson_line_start["line"] == pyright_line_start["line"]
+            # pyrights offsets are zero-based
+            assert rdjson_line_start["line"] == pyright_line_start["line"] + 1
             assert rdjson_line_start["column"] == pyright_line_start["character"]
```
Comment (jordemort) [inline pyright_to_rdjson/tests/test_pyright_to_rdjson.py:48]:
```suggestion
            assert rdjson_line_start["column"] == pyright_line_start["character"] + 1
```

File: pyright_to_rdjson/tests/test_pyright_to_rdjson.py [hunk 4]
```diff
@@ -43,8 +43,9 @@ def test_pyright_to_rdjson(self):
...
 
-            assert rdjson_line_end["line"] == pyright_line_end["line"]
+            assert rdjson_line_end["line"] == pyright_line_end["line"] + 1
             assert rdjson_line_end["column"] == pyright_line_end["character"]
```
Comment (jordemort) [inline pyright_to_rdjson/tests/test_pyright_to_rdjson.py:51]:
```suggestion
            assert rdjson_line_end["column"] == pyright_line_end["character"] + 1
```

**Comment by gersmann** [comment]
makes sense, thanks for checking this out

**Comment by jordemort** [comment]
Merged, thanks!

**Pull request merged**